contexts.go

```diff
@@ -0,0 +1,30 @@
+package config
+
+import (
+	"context"
+	"fmt"
+)
+
+type configKeyType struct{}
+
+var configKey = configKeyType{}
+
+func WithContext(ctx context.Context, cfg *Config) context.Context {
```

Comment:
Is this name odd? `WithContext` usually takes a `ctx` and an `X` and returns a modified `X` with the context applied, no?

Reply:
Suggestion (or something simliar):

```go
	// Add default services to context
	ctx = process.AddHealthToContext(ctx, health)
	ctx = log.AddToContext(ctx, logger)
	ctx = service.AddToContext(ctx, serviceContainer)
	ctx = config.AddToContext(ctx, cfg)
```

Reply:
I guess it didn't take long to find a few options.

It looks like the `http` lib has a `WithContext` on `http.Request` that returns a request, but there's also `context.WithValue` which returns the parent context with the value included. There are also a few of the other `context.WithX` methods that use that pattern too. That might be why I went with that convention.

Given that, how would you feel about something like `WithConfig(context.Context, *Config) context.Context` (and similar in the other packages)? I'm hesitant to go with `AddXToContext` because technically it's not adding it to the context, it's returning a new one with that value included.

Reply:
Another option that briefly popped into my brain that I wasn't a big fan of but I figured I'd mention as well is maybe something like `InContext` or `HealthInContext`?

Reply:
https://sourcegraph.com/search?q=context:global+lang:Go+func+WithContext&patternType=lucky

`WithConfig`, etc, would be perfect! I just wanted to comment on the `WithContext`, which seems to in some of the standard lib (you mentioned `http.Request` as one) to be modifying some other value _with_ a context, not the other way around.

Reply:
Sounds good; I'll update the other packages to use that naming convention!

Reply:
Would we still want to use just `FromContext` now that the "include" methods will have the name of the thing being included? I'm leaning toward yes and leaving those methods named the same.

Reply:
Yeah unless it's ambiguous (e.g., `HealthFromContext`) but I think you've already gotten those cases.